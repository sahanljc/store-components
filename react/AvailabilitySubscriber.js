export { default } from './components/AvailabilitySubscriber'
