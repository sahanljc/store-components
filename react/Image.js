import Image from './components/Image/index'

export default Image
