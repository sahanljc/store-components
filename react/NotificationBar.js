import NotificationBar from './components/NotificationBar/index'

export default NotificationBar
