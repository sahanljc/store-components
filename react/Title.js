import Title from './components/Title/index'

export default Title
