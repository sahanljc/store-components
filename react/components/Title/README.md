# Title

## Description

`Title` is a VTEX Component that shows the given text input as a title.
This Component can be imported and used by any VTEX App.

:loudspeaker: **Disclaimer:** Don't fork this project; use, contribute, or open issue with your feature request.

## Table of Contents
- [Usage](#usage)
  - [Blocks API](#blocks-api)
  - [Configuration](#configuration)
  - [Styles API](#styles-api)

## Usage

You should follow the usage instruction in the main [README](https://github.com/vtex-apps/store-components/blob/master/README.md#usage).

Then, add `title` block into our app theme. 

### Blocks API

This component has an interface that describes which rules must be implemented by a block when you want to use the `Title`.

```json
  "title": {
    "component": "Title"
  }
```

### Configuration

Through the Storefront, you can change the `Title`'s behavior and interface. However, you also can make in your theme app, as Dreamstore does.

| Prop name | Type | Description |
| --- | --- | --- |
| `content` | `String` | Text to be displayed |
| `alignment` | `String` | Title alignment (left, right or center) |

### Styles API
:construction: :construction: :construction: