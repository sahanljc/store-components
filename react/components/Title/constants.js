export const LEFT = 'left'
export const RIGHT = 'right'
export const CENTER = 'center'

export const LEVELS = [1, 2, 3, 4, 5, 6]
