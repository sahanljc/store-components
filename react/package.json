{
  "private": true,
  "scripts": {
    "pretest": "yarn",
    "test": "vtex-test-tools test",
    "lint": "eslint --ext js,jsx,ts,tsx ."
  },
  "dependencies": {
    "@material-ui/core": "^3.5.1",
    "@vtex/css-handles": "^1.0.2",
    "apollo-client": "^2.5.1",
    "classnames": "^2.2.5",
    "debounce": "^1.2.0",
    "downshift": "^1.31.14",
    "graphql": "~0.13.2",
    "graphql-tag": "~2.8.0",
    "prop-types": "^15.6.2",
    "ramda": "^0.25.0",
    "react": "^16.3.2",
    "react-apollo": "~2.5.2",
    "react-content-loader": "^3.1.2",
    "react-dom": "^16.8.3",
    "react-html-parser": "^2.0.2",
    "react-id-swiper": "^1.6.8",
    "react-intl": "^2.8.0",
    "react-resize-detector": "^3.0.1",
    "react-share": "^2.1.1",
    "react-slick": "^0.23.1",
    "sanitize-html": "^1.20.0",
    "slick-carousel": "^1.8.1"
  },
  "devDependencies": {
    "@vtex/test-tools": "^0.1.2",
    "apollo-cache-inmemory": "^1.4.3",
    "eslint": "^5.16.0",
    "eslint-config-vtex-react": "^4.0.0",
    "prettier": "^1.16.4",
    "typescript": "^3.4.1"
  },
  "vtexTestTools": {
    "defaultLocale": "en"
  }
}
