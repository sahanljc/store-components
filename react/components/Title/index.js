import PropTypes from 'prop-types'
import React, { Component } from 'react'

import { LEFT, RIGHT, CENTER } from './constants.js'
import title from './title.css'

/**
 * Name component. Show name and relevant SKU information of the Product Summary
 */
class Title extends Component {
  static propTypes = {
    /** Text to be displayed */
    content: PropTypes.string,
    /** Alignment of the title */
    alignment: PropTypes.oneOf([LEFT, RIGHT, CENTER]).isRequired,
  }

  static defaultProps = {
    alignment: CENTER,
  }

  render() {
    const { content, alignment } = this.props

    return (
      <div className={`${title.titleContainer} flex justify-between`}>
        <div className={`${title.titleBox} w-100 ma1`}>
          {alignment == LEFT && <h3 className="tl">{content}</h3>}
          {alignment == CENTER && <h3 className="tc">{content}</h3>}
          {alignment == RIGHT && <h3 className="tr">{content}</h3>}
        </div>
      </div>
    )
  }
}

Title.schema = {
  title: 'editor.title.title',
  description: 'editor.title.description',
  type: 'object',
  properties: {
    content: {
      type: 'string',
      title: 'editor.title.content',
      isLayout: false,
    },
    alignment: {
      type: 'string',
      title: 'editor.title.alignment',
      enum: [LEFT, RIGHT, CENTER],
      enumNames: [
        'editor.title.alignment.left',
        'editor.title.alignment.right',
        'editor.title.alignment.center',
      ],
      default: CENTER,
      widget: {
        'ui:widget': 'radio',
        'ui:options': {
          inline: true,
        },
      },
      isLayout: true,
    },
  },
}

export default Title
