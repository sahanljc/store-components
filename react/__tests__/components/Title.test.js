import React from 'react'
import { render } from '@vtex/test-tools/react'

import Title from './../../Title'

describe('<Title /> component', () => {
  const renderComponent = customProps => {
    const props = {
      content: 'title',
      ...customProps,
    }
    const comp = <Title {...props} />

    return render(comp)
  }

  it('should be rendered', () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toBeTruthy()
  })

  it('should match snapshot with default center alignment', () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })

  it('should match snapshot with left alignment', () => {
    const { asFragment } = renderComponent({ alignment: 'left' })
    expect(asFragment()).toMatchSnapshot()
  })

  it('should match snapshot with right alignment', () => {
    const { asFragment } = renderComponent({ alignment: 'right' })
    expect(asFragment()).toMatchSnapshot()
  })

  it('should match snapshot with <h1>', () => {
    const { asFragment } = renderComponent({ level: 1 })
    expect(asFragment()).toMatchSnapshot()
  })

  it('should match snapshot with <h6> with right alignment', () => {
    const { asFragment } = renderComponent({ level: 6, alignment: 'right' })
    expect(asFragment()).toMatchSnapshot()
  })
})
