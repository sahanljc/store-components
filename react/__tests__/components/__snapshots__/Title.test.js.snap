// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> component should match snapshot with default center alignment 1`] = `
<DocumentFragment>
  <div
    class="titleContainer flex justify-between"
  >
    <div
      class="titleBox w-100 ma1"
    >
      <h3
        class="tc"
      >
        title
      </h3>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Title /> component should match snapshot with left alignment 1`] = `
<DocumentFragment>
  <div
    class="titleContainer flex justify-between"
  >
    <div
      class="titleBox w-100 ma1"
    >
      <h3
        class="tl"
      >
        title
      </h3>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Title /> component should match snapshot with right alignment 1`] = `
<DocumentFragment>
  <div
    class="titleContainer flex justify-between"
  >
    <div
      class="titleBox w-100 ma1"
    >
      <h3
        class="tr"
      >
        title
      </h3>
    </div>
  </div>
</DocumentFragment>
`;
